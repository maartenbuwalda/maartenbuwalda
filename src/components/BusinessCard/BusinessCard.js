import React from 'react'
import PropTypes from 'prop-types'
import Card from '../../components/Card'
import Logo from '../../components/Logo'
import styled from 'styled-components'
import colors from '../../constants/colors'
import { Link } from 'react-router-dom'

const CardLayout = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  max-height: 16rem;
  padding: 1rem;
  margin: 1rem;

  a {
    text-decoration: none;
  }
`

const TitleWrapper = styled.div`
  h1, p {
    margin: 0;
    text-align: center;
    color: ${colors.fadedWhite};
  }
  h1 {
    font-size: 2rem;
    margin-bottom: .25rem;
  }
  p {
    text-transform: uppercase;
    letter-spacing: 1px;
    font-size: .75rem;
  }
`

const Contact = styled.div`
  border-top: 1px solid ${colors.gray};
  padding-top: 1rem;
  display: flex;
  justify-content: center;
  align-items: center;
  list-style: none;
  margin: 1rem 0 0;

  a {
    margin: 0 .5rem;
  }
`

const StyledLogo = styled(Logo)`
  margin: auto;
  margin-bottom: 1rem;
  padding: 0;
`

const BusinessCard = ({ position }) => {
  return (
    <Card background="colored" position={position}>
      <CardLayout>
        <Link to="/">
          <StyledLogo white animated/>
          <TitleWrapper>
            <h1>Maarten Buwalda</h1>
            <p>front-end developer</p>
          </TitleWrapper>
        </Link>
        <Contact>
          <a target="_blank" rel="noopener noreferrer" href="https://www.linkedin.com/in/maartenbuwalda/">linkedin</a>-
          <a target="_blank" rel="noopener noreferrer" href="https://github.com/maartenbuwalda">github</a>-
          <a target="_blank" rel="noopener noreferrer" href="mailto:maarten@maartenbuwalda.com">email</a>
        </Contact>
      </CardLayout>
    </Card>
  )
}

BusinessCard.propTypes = {
  position: PropTypes.string,
}

export default BusinessCard
