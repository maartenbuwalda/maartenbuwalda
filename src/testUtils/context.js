const context = {
  theme: {},
  updateTheme: jest.fn(),
}

export default context
