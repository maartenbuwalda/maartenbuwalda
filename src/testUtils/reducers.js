export const reducerReturnsInitialState = (reducer, initialState) => {
  it('should succeed: return the initialState', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })
}
