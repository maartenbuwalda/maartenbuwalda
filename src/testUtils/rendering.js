import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import 'jest-styled-components'

export const componentRenders = component => {
  it('renders correctly', () => {
    const shallowed = shallow(component)
    expect(toJson(shallowed)).toMatchSnapshot()
  });
}
