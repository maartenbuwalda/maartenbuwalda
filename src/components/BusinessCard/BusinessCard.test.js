import React from 'react'
import BusinessCard from './index'
import { componentRenders } from '../../testUtils/rendering'

describe('BusinessCard component', () => {
  const component = <BusinessCard />
  componentRenders(component)
})
