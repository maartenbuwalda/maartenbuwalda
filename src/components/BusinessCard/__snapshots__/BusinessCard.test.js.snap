// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessCard component renders correctly 1`] = `
<Card
  background="colored"
  content=""
>
  <styled.div>
    <Link
      replace={false}
      to="/"
    >
      <Styled(Logo)
        animated={true}
        white={true}
      />
      <styled.div>
        <h1>
          Maarten Buwalda
        </h1>
        <p>
          front-end developer
        </p>
      </styled.div>
    </Link>
    <styled.div>
      <a
        href="https://www.linkedin.com/in/maartenbuwalda/"
        rel="noopener noreferrer"
        target="_blank"
      >
        linkedin
      </a>
      -
      <a
        href="https://github.com/maartenbuwalda"
        rel="noopener noreferrer"
        target="_blank"
      >
        github
      </a>
      -
      <a
        href="mailto:maarten@maartenbuwalda.com"
        rel="noopener noreferrer"
        target="_blank"
      >
        email
      </a>
    </styled.div>
  </styled.div>
</Card>
`;
